src/testdir/test_registers.vim

```diff
@@ -164,6 +165,19 @@ func Test_register_one()
   bwipe!
 endfunc
 
+func Test_recording_status_in_ex_line()
+  norm qx
+  redraw!
+  call assert_equal('recording @x', Screenline(&lines))
+  set shortmess=q
+  redraw!
```

Comment:
With "redraw" instead of "redraw!", the assert below would fail:
```
Expected 'recording' but got 'recording @x'
```
Should this be considered as a bug?

Reply:
:redraw without ! only redraws text that has changed, and an option value change isn't checked for.
I think that's OK.  Would be inefficient to check for lots of things that might have changed but usually didn't.